### DISCUSSION TITLE
Open Url and access CLI options from Firefox CLI

### DISCUSSION THREAD

**Discussion by lamyergeier** [Homebrew/discussions#3333] [Casks]
I have installed firefox using brew cask on macOS 14. It got installed at /Applications/Firefox.app/Contents/MacOS/firefox.

I can access CLI options like `/Applications/Firefox.app/Contents/MacOS/firefox -h`

But it doesn't take any value. For example following doesn't work:

```bash
/Applications/Firefox.app/Contents/MacOS/firefox  --new-tab duckduckgo.com
/Applications/Firefox.app/Contents/MacOS/firefox  duckduckgo.com
```

What can I do?

**Comment by alebcay** [comment]
Based on [this Stack Overflow answer](https://stackoverflow.com/questions/10866628/firefox-openurl-osx-command-line), you can probably do:
```
open -a Firefox https://duckduckgo.com [--args <any additional args to pass to Firefox>]
```

Note: The `http[s]://` must not be omitted or else this will not work.

**Comment by lamyergeier** [comment]
@alebcay  This does not work: `open -a firefox --new-instance https://google.com`

**Reply by alebcay** [reply]
Try `open -a firefox https://google.com --args --new-instance` (note that the URL comes _before_ any additional arguments; moreover, note that any Firefox-specific arguments come _after_ `--args`).

No guarantees that it works - this may be something to ask the folks at Firefox instead. I imagine that some of the CLI arguments for Windows/Linux may not work the same way or may not be available at all on macOS.

Refer also to the previously linked Stack Overflow page for additional examples that other people confirm are working (I tried one or two from that thread and they seemed to work, although I realize you may be trying to achieve something specific).
